src/plugin/cesium/sync/featureconverter.js

```diff
@@ -540,11 +540,18 @@ plugin.cesium.sync.FeatureConverter.prototype.olLineStringGeometryToCesium = fun
  */
 plugin.cesium.sync.FeatureConverter.prototype.createLinePrimitive = function(positions, context, style, opt_type) {
   var type = opt_type || 'PolylineGeometry';
-  var appearance = new Cesium.PolylineColorAppearance();
```

Comment:
This is why the color changes stopped working. `Cesium.PolylineMaterialAppearance` uses the color on the geometry instance attributes, which is what we update in `updatePrimitive` when the color changes. The new code is using `Cesium.PolylineMaterialAppearance` with a material that has its own `color` property, so the material would need to be updated instead.

I would propose the following to fix this without having to handle both of these cases going forward:

- Change `createColoredPrimitive` to use `Cesium.MaterialAppearance` as the `appearance`, setting `options.material = Cesium.Material.fromType(Cesium.Material.ColorType, {color: color})`. This will make the default material for lines/polygons use the same type of appearance as dashed lines, so they can be updated in the same fashion.
- Remove the `color` param from `createGeometryInstance`. We will no longer be using color on the geometry instance attributes.
- Change `updatePrimitive` to update the color on the material. It should no longer need to get the attributes with `getGeometryInstanceAttributes`, and instead get the appearance/material and update that.
- Test lines, polygons, extruded polygons (load a KML with extrude), ellipses/ellipsoids, etc to make sure color/selection/highlight still works everywhere.

Reply:
Did you mean `Cesium.PolylineMaterialAppearance ` instead of `Cesium.MaterialAppearance` ?

Reply:
No, `createColoredPrimitive` is used for creating polygons so it should just use `MaterialAppearance`. I believe all of the polyline paths have their own appearance.

Reply:
I set it as `PolylineMaterialAppearance` and it seems to be working ... tried with just `MaterialAppearance` and can't tell the different actually, so I can stick with the `MaterialAppearance`

Reply:
Sorry my idea was incorrect, and I didn't notice because we usually render polygons with polylines so their line width will be correct on Windows.

I tested with extruded polygons, which goes down the `olPolygonGeometryToCesium` path and `MaterialAppearance` does not work there. You can test this with extruded polygons. We may have to leave these functions as they currently are, and `updatePrimitive` will need to determine how to update the color appropriately.

Reply:
Okay, I've reverted those changes